express-notes-api/index.js

```diff
@@ -0,0 +1,120 @@
+const express = require('express');
+const fs = require('fs');
+const data = require('./data.json');
+const app = express();
+
+app.use(express.json());
+
+const badRequestNegInt = { error: 'id must be a positive integer' };
+const badRequestNoContent = { error: 'content is a required field' };
+const unexpectedError = { error: 'An unexpected error occured.' };
+
+app.post('/api/notes', (req, res) => {
+  const note = req.body;
+  if (Object.keys(note).length === 0) {
+    res.status(400);
+    res.json(badRequestNoContent);
+  } else {
+    note.id = data.nextId;
+    data.notes[data.nextId] = note;
+    data.nextId++;
+    const parseData = JSON.stringify(data, null, 2);
+    fs.writeFile('data.json', parseData, err => {
+      if (err) {
+        // eslint-disable-next-line no-console
+        console.log(err);
+        res.status(500);
+        res.send(unexpectedError);
+      }
+      // eslint-disable-next-line no-console
+      console.log('New note added.');
```

Comment:
We'll want to include an `else` statement to ensure that the server doesn't try to respond to same request more than once.